src/onedrive.d

```diff
@@ -1012,10 +1012,12 @@ final class OneDriveApi
 		auto expirationDateTime = Clock.currTime(UTC()) + subscriptionExpirationInterval;
 		const(char)[] url;
 		url = subscriptionUrl;
+		// Create a resource item using the account driveId rather than using /me/drive/root
```

Comment:
This only works when driveId is set in the configuration. I suggest this code which works well for both personal, business and sharepoint drives :

```
string resourceItem;
if (driveId.length) {
        resourceItem = "/drives/" ~ driveId ~ "/root";
} else {
        resourceItem = "/me/drive/root";
}

```

Reply:
This would have been the updated PR once feedback from you. I will update the PR with the above.